examples/api/test/material/icon_button/icon_button.3_test.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/icon_button/icon_button.3.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('It should select and unselect the icon buttons', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.IconButtonToggleApp(),
+    );
+    expect(find.widgetWithIcon(IconButton, Icons.settings_outlined), findsExactly(8));
+    final Finder unselectedIconButtons = find.widgetWithIcon(IconButton, Icons.settings_outlined);
+    for (int i = 0; i <= 6; i += 2) {
+      expect(tester.widget<IconButton>(unselectedIconButtons.at(i)).onPressed, isA<VoidCallback>());
+      expect(tester.widget<IconButton>(unselectedIconButtons.at(i + 1)).onPressed, isNull);
+    }
+    expect(tester.widgetList<IconButton>(unselectedIconButtons).map((IconButton iconButton) => iconButton.isSelected), everyElement(isFalse));
```

Comment:
```suggestion
    for (int i = 0; i <= 6; i++) {
      final IconButton button = tester.widget<IconButton>(unselectedIconButtons.at(i));
      expect(button.onPressed, i.isEven ? isA<VoidCallback>() : isNull);
      expect(button.isSelected, isFalse);
    }
```
Do you think this alternative is more readable? I prefer it because it avoids repeating the long conversion to `IconButton` and the mix of `for` and `map`. Up to you to adopt it or not.

Reply:
Sure, it looks simpler, I modified it in [refactor: Better readable code](https://github.com/flutter/flutter/pull/149988/commits/28a2fe351bc06f4089dd716c32aab24f53dab5d8)

Reply:
> Sure, it looks simpler, I modified it in [refactor: Better readable code](https://github.com/flutter/flutter/pull/149988/commits/28a2fe351bc06f4089dd716c32aab24f53dab5d8)

Great! Can you apply the same to the other similar sections?

Reply:
Oops sorry, I pushed it in [format: Better formatting](https://github.com/flutter/flutter/pull/149988/commits/543e262b3df9fd24c856e356c055cfa433738c8b)